app/src/main/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModel.kt

```diff
@@ -35,7 +36,17 @@ class ProfileViewModel @Inject constructor(
 		}
 	}
 
+	fun updateAvatar(uri: Uri) {
+		viewModelScope.launch {
+			_profile.value = Loading
+			authRepository.updateUserAvatar(uri)
+			fetchProfile()
+		}
+	}
+
 	private fun fetchProfile() {
-		_profile.value = Loaded(authRepository.userProfile())
+		viewModelScope.launch {
+			_profile.value = Loaded(authRepository.userProfile())
```

Comment:
maybe you could use `runCatching` block and add TODO in case on fail, something like this:
```runCatching { authRepository.userProfile() }
            .onSuccess { _profile.value = Loaded(it) }
            .onFailure { // TODO to be done }```
then it's clear you didn't forget about error handling and it will be done later?

Reply:
Thanks for advice 😄